editor/components/colors/with-colors.js

```diff
@@ -1,91 +1,134 @@
+ *                                  the color context is computed by applying a kebab case transform to the value.
+ *                                  Color context represents the context/place where the color is going to be used.
+ *                                  The class name of the color is generated using 'has' followed by the color name
+ *                                  and ending with the color context all in kebab case e.g: has-green-background-color.
+ *
  *
  * @return {Function} Higher-order component.
  */
-export default ( mapGetSetColorToProps ) => createHigherOrderComponent(
-	compose( [
-		withSelect(
-			( select ) => {
+export default ( ...args ) => {
+	if ( isFunction( args[ 0 ] ) ) {
+		deprecated( 'Using withColors( mapGetSetColorToProps ) ', {
+			version: '3.2',
+			alternative: 'withColors( colorAttributeName, { secondColorAttributeName: \'color-context\' }, ... )',
+		} );
+		return withColorsDeprecated( args[ 0 ] );
+	}
+
+	const colorMap = reduce( args, ( colorObj, arg ) => {
+		return {
+			...colorObj,
+			...( isString( arg ) ? { [ arg ]: kebabCase( arg ) } : arg ),
+		};
+	}, {} );
+
+	return createHigherOrderComponent(
+		compose( [
+			withSelect( ( select ) => {
 				const settings = select( 'core/editor' ).getEditorSettings();
 				return {
 					colors: get( settings, [ 'colors' ], DEFAULT_COLORS ),
 				};
 			} ),
-		( WrappedComponent ) => {
-			return class extends Component {
-				constructor() {
-					super( ...arguments );
-					/**
-					* Even though, we don't expect setAttributes or colors to change memoizing it is essential.
-					* If setAttributes or colors are not memoized, each time memoizedGetColor/memoizedSetColor are called:
-					* a new function reference is returned (even if arguments have not changed).
-					* This would make our memoized chain useless.
-					*/
-					this.memoizedGetColor = memoize( this.memoizedGetColor, { maxSize: 1 } );
-					this.memoizedSetColor = memoize( this.memoizedSetColor, { maxSize: 1 } );
-				}
+			( WrappedComponent ) => {
+				return class extends Component {
+					constructor( props ) {
+						super( props );
+
+						this.setters = this.createSetters();
 
-				memoizedGetColor( colors ) {
-					return memoize(
-						( colorName, customColorValue, colorContext ) => {
-							return {
-								name: colorName,
-								class: getColorClass( colorContext, colorName ),
-								value: getColorValue( colors, colorName, customColorValue ),
-							};
-						}
-					);
-				}
+						this.state = {};
+					}
 
-				memoizedSetColor( setAttributes, colors ) {
-					return memoize(
-						( colorNameAttribute, customColorAttribute ) => {
-							return setColorValue( colors, colorNameAttribute, customColorAttribute, setAttributes );
-						}
-					);
-				}
+					createSetters() {
+						return reduce( colorMap, ( settersAcc, colorContext, colorAttributeName ) => {
+							const ufColorAttrName = upperFirst( colorAttributeName );
```

Comment:
`ufColorAttrName` is a lot of abbreviation–I'm not sure what it means and I tried to read above but it was hard to tell from context.